### ISSUE TITLE
Add back proxy support

### ISSUE THREAD

**Issue by cldwalker** [LightTable/LightTable#1984]
Proxy support [was disabled](https://github.com/LightTable/LightTable/blob/dec215a2846644e5544c971c4275f6d8313ad730/src/lt/objs/deploy.cljs#L193) and not implemented in the move from node-webkit to electron. Updating [the old proxy](https://github.com/LightTable/LightTable/blob/dec215a2846644e5544c971c4275f6d8313ad730/src/lt/objs/deploy.cljs#L29-L48) with [the electron one](https://github.com/atom/electron/blob/1bc49487add140f22ea4e454dcb0050e08679b4b/docs/api/app.md#appresolveproxyurl-callback) seems straightforward. What is currently unknown is how to verify this works. For anyone who is behind a proxy, please provide a means for the core team to reproduce being behind a proxy so we can implement and QA this

**Comment by EarthCitizen** [comment]
For anyone looking for temporary solution:

The issue I found was that it was not able to connect through NTLM authenticated proxies. The workaround for me was to install and use Cntlm. It is a local proxy that connects to your network's proxy on your behalf. So, a proxy to your proxy. You give it the credentials to your network's proxy then just point LightTable at localhost.

**Comment by wodin** [comment]
I think the only way to reproduce being behind a proxy would be to be behind a proxy :)
Maybe running a simple proxy on localhost would be enough?
e.g. something like this:
http://stackoverflow.com/a/7271863/495319
or maybe this:
https://github.com/nodejitsu/node-http-proxy

**Comment by EarthCitizen** [comment]
I think specifically it was failing behind NTLM proxies, correct?

On Sunday, October 25, 2015, Michael Wood <notifications@github.com> wrote:

> I think the only way to reproduce being behind a proxy would be to be
> behind a proxy :)
> Maybe running a simple proxy on localhost would be enough?
> e.g. something like this:
> http://stackoverflow.com/a/7271863/495319
> or maybe this:
> https://github.com/nodejitsu/node-http-proxy
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LightTable/LightTable/issues/1984#issuecomment-150900177>
> .
>


-- 
"Simplicity is the ultimate sophistication."
 --  Leonardo da Vinci

**Comment by kenny-evitt** [comment]
[Squid](http://www.squid-cache.org/) might be a good candidate for something that could be setup so that a contributor could reproduce the issue(s).

@EarthCitizen Care to figure out how to setup a proxy server and reproduce the problems you described in #1895?

**Comment by EarthCitizen** [comment]
Does Squid support NTLM authentication?

On Wednesday, October 28, 2015, kenny-evitt <notifications@github.com>
wrote:

> Squid <http://www.squid-cache.org/> might be a good candidate for
> something that could be setup so that a contributor could reproduce the
> issue(s).
>
> @EarthCitizen <https://github.com/EarthCitizen> Care to figure out how to
> setup a proxy server and reproduce the problems you described in #1895
> <https://github.com/LightTable/LightTable/issues/1895>?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LightTable/LightTable/issues/1984#issuecomment-152045896>
> .
>


-- 
"Simplicity is the ultimate sophistication."
 --  Leonardo da Vinci

**Comment by kenny-evitt** [comment]
@EarthCitizen yup

**Comment by EarthCitizen** [comment]
If there is a build ready, I can try it at work, which is where I had the
problem.

On Wednesday, October 28, 2015, kenny-evitt <notifications@github.com>
wrote:

> @EarthCitizen <https://github.com/EarthCitizen> yup
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LightTable/LightTable/issues/1984#issuecomment-152068940>
> .
>


-- 
"Simplicity is the ultimate sophistication."
 --  Leonardo da Vinci

**Comment by kenny-evitt** [comment]
@EarthCitizen A LT build? There's the recent 0.8.0-alpha release we just published but there haven't been any changes made for proxy support. There are binaries for Squid at the link in my previous comment but we don't need you to reproduce your issue. We need help figuring out how to setup a proxy so anyone else can reproduce it.

**Comment by wodin** [comment]
Squid does indeed support ntlm in conjunction with Samba. I've never tried it, though.

Here's some old documentation about setting it up:

http://wiki.squid-cache.org/ConfigExamples/Authenticate/Ntlm

**Comment by huiwang** [comment]
I'm behind a proxy and i confirm that it doesn't work neither with the latest version 0.8.0.

You can mark me as the tester.

**Comment by scelera** [comment]
@EarthCitizen - How did you point LightTable to your localhost?  I have CNTLM running on my system but haven't found anything that lets me point LightTable to it?

**Comment by EarthCitizen** [comment]
@scelera If I remember correctly, I believe I did it with the http_proxy and https_proxy environment variables

**Comment by scelera** [comment]
@EarthCitizen Thanks.  Unfourtuneatly for me I already have those set and it still doesn't work.

**Comment by EarthCitizen** [comment]
Make sure you have both set:

http_proxy=http://localhost:<your port>
https_proxy=https://localhost:<your port>

Also,

**Comment by scelera** [comment]
Yup, both are set.  I've been using CNTLM for over a year now.  Works flawlessly with everything else.  I'm on LightTable 0.8.0 if that makes a difference.

**Comment by EarthCitizen** [comment]
The only other thing I can think of is to try the various http.proxy___ Java system properties

**Comment by scelera** [comment]
Thanks.  I think I got that part figured out.  I updated the tunnel.js file in \resources\app\core\node_modules\request.  Now though it's giving me this error

TypeError: lt.objs.deploy.request.get is not a function
    at lt.objs.deploy.download_file (file:///C:/usryrs/lighttable-0.8.0-windows/resources/app/core/node_modules/lighttable/bootstrap.js:28950:33)
    at lt.objs.plugins.download_metadata (file:///C:/usryrs/lighttable-0.8.0-windows/resources/app/core/node_modules/lighttable/bootstrap.js:29393:39)
    at Function.lt.objs.plugins.__BEH__check_local_metadata_cache (file:///C:/usryrs/lighttable-0.8.0-windows/resources/app/core/node_modules/lighttable/bootstrap.js:29822:419)
    at c (file:///C:/usryrs/lighttable-0.8.0-windows/resources/app/core/node_modules/lighttable/bootstrap.js:6196:14)
    at a (file:///C:/usryrs/lighttable-0.8.0-windows/resources/app/core/node_modules/lighttable/bootstrap.js:6236:18)
    at c (file:///C:/usryrs/lighttable-0.8.0-windows/resources/app/core/node_modules/lighttable/bootstrap.js:20446:76)
    at a (file:///C:/usryrs/lighttable-0.8.0-windows/resources/app/core/node_modules/lighttable/bootstrap.js:20480:18)
    at a (file:///C:/usryrs/lighttable-0.8.0-windows/resources/app/core/node_modules/lighttable/bootstrap.js:20491:34)
    at b (file:///C:/usryrs/lighttable-0.8.0-windows/resources/app/core/node_modules/lighttable/bootstrap.js:20495:14)
    at file:///C:/usryrs/lighttable-0.8.0-windows/resources/app/core/node_modules/lighttable/bootstrap.js:29387:28

Is this releated to the proxy problem or is this something else?

**Comment by kenny-evitt** [comment]
@scelera That looks new.

**Comment by olange** [comment]
Same issue for me: I cannot refresh the plugins list, nor update, nor install plugins, being behind my company's proxy. I'm using LightTable 0.8.0 / Windows 7 Enterprise / Java JDK 7.

LightTable's console shows following error when trying to refresh the plugins list:

```clojure
TypeError: Request path contains unescaped characters.
    at new ClientRequest (_http_client.js:51:11)
    at TunnelingAgent.exports.request (http.js:31:10)
    at TunnelingAgent.createSocket (E:\projects\lighttable\resources\app\core\node_modules\request\tunnel.js:117:25)
    at TunnelingAgent.createSecureSocket [as createSocket] (E:\projects\lighttable\resources\app\core\node_modules\request\tunnel.js:184:41)
    at TunnelingAgent.addRequest (E:\projects\lighttable\resources\app\core\node_modules\request\tunnel.js:80:8)
    at new ClientRequest (_http_client.js:133:16)
    at Object.exports.request (http.js:31:10)
    at Object.exports.request (https.js:163:15)
    at Request.start (E:\projects\lighttable\resources\app\core\node_modules\request\main.js:559:30)
    at Request.end (E:\projects\lighttable\resources\app\core\node_modules\request\main.js:1075:28)
```

I'm starting LightTable with this script:

```posh
@echo off
echo.

:set-java-home
SET "JAVA_HOME=…\jboss\jdk\64\1.7.0_51"
set "JAVA_CMD=%JAVA_HOME%\bin\java"
ECHO   Set JAVA_HOME to "%JAVA_HOME%" ...
ECHO   Set JAVA_CMD to "%JAVA_CMD%" ...

:set-http-proxy
SET "HTTP_PROXY=http://proxy:8080"
SET "HTTPS_PROXY=http://proxy:8080"
SET "HTTP_NO_PROXY=*.mycompany.com|localhost"
echo   Set HTTP_PROXY and HTTPS_PROXY to %HTTP_PROXY% ...
ECHO   Set HTTP_NO_PROXY to "%HTTP_NO_PROXY%" ...

:launch
echo   Launching Light Table ...
set "LTHOME=E:\projects\lighttable"
start %LTHOME%\LightTable.exe
```

And LightTable's Node configuration looks like this:

```ini
E:\projects\lighttable\resources\app\core\node_modules\replace>npm config list
; cli configs
user-agent = "npm/3.7.3 node/v5.9.0 win32 x64"

; userconfig C:\Users\MyUserName\.npmrc
https-proxy = "http://proxy:8080/"
proxy = "http://proxy:8080/"

; builtin config undefined
prefix = "C:\\Users\\MyUserName\\AppData\\Roaming\\npm"

; node bin location = C:\Program Files\nodejs\node.exe
; cwd = E:\projects\lighttable\resources\app\core\node_modules\replace
; HOME = C:\Users\MyUserName
```

**Comment by olange** [comment]
Updating the `request` module from version 2.14 to version 2.72 (latest as of today), then restarting LightTable, solved the issue for me. I can now refresh the plugins list and install new ones, while being behind my company's proxy:

```posh
C:> cd /d E:\projects\lighttable\resources\app\core
E:\projects\lighttable\resources\app\core> npm list request
LightTable@0.8.1 E:\projects\lighttable\resources\app\core
`-- request@2.14.0
E:\projects\lighttable\resources\app\core>npm update request
[… snip snip …]
LightTable@0.8.1 E:\projects\lighttable\resources\app\core
`-- request@2.72.0
``` 

I followed @geisi93 's suggestion in #321 (using `npm update request` rather than updating `tunnel-agent` manually in `request`'s dependencies).

**Comment by kenny-evitt** [comment]
@olange If that really is all we need to do, that's fantastic!

Mind submitting a PR with that change? If you do, I'll try to get to it today.

**Comment by howardlinus** [comment]
yes, it works for me. Many thanks

**Comment by Armaklan** [comment]
Update request work for me.
Win 7. Lt launch on "git bash" command after export http_proxy & https_proxy.

**Comment by mluisbrown** [comment]
I'm getting the same issue as @howardlinus after updating the `request` module as per @olange 's instructions. What I find is that there is a `metadata_temp.tar.gz` of 0 bytes in the `...\AppData\Roaming|LightTable` directory (which disappears when I quite LightTable). This suggests to me that something in the proxy is still preventing the metadata file from being downloaded correctly, so the 0 byte file expands to nothing, whereas it should expand to the `metadata_temp` directory.

In my case I'm pretty sure that this is my work proxy that is causing the problem. I'm using a local proxy (Fiddler) which does the same thing as Cntlm (proxies requests to the real proxy using NTLM auth). I can see in the logs for the proxy that LightTable is accessing `api.github.com:443` before the error occurs. My work proxy inserts it's own SSL certificate in the chain (yes, man in the middle) to be able to monitor SSL traffic (common practice in banks). This doesn't play well with a lot of services, GitHub among them. I fear that the only solution, at least in my specific case, would be a way to configure LightTable to not use SSL when communicating with GitHub.

**Comment by kenny-evitt** [comment]
@mluisbrown [I think you're out of luck if you can't access GitHub over HTTPS](https://developer.github.com/v3/#schema):

> All API access is over HTTPS, and accessed from the `https://api.github.com`. All data is sent and received as JSON.

Would you please create a separate issue as this issue just covers generic proxy support.

**Comment by mluisbrown** [comment]
@kenny-evitt I've looked at this further, and I don't think that my work proxy is blocking SSL access to api.github.com after all (I can access it via a browser no problem). Still, for whatever reason the `metadata-temp.tar.gz` file that's received from GitHub is 0 bytes long, which leads to the `Error: ENOENT: no such file or directory, scandir 'C:\Users\myid\AppData\Roaming\LightTable\metadata-temp'` error mentioned above.

I can download https://api.github.com/repos/LightTable/plugin-metadata/tarball/master ok in my browser, so there's no reason that LightTable shouldn't be able to download it either.

**Comment by mluisbrown** [comment]
More info. My work proxy is not *blocking* SSL access, but they are inserting their own SSL certificate in the chain in order to be able to monitor SSL traffic of employees as I mentioned above. This means that software that performs certificate verification will fail: 

`curl -i https://api.github.com/users/octocat/orgs` fails with an SSL certificate problem. It only works if I use the `--insecure` option. I imagine that [request](https://www.npmjs.com/package/request) also fails for the same reason. I will verify that and see if there's any option that can be passed to `request` for it to work in "insecure" mode.

The reason it works fine in the browsers at work is because they have been pre-configured to trust the company certificate :-)

**Comment by mluisbrown** [comment]
Confirmed that it's the SSL certificate problem. The following node script:

```javascript
var request = require('request');
var fs = require('fs');

request({
    url: 'https://api.github.com/repos/LightTable/plugin-metadata/tarball/master',
    strictSSL: false,
    headers: { 'User-Agent': 'Light Table'}
}).pipe(fs.createWriteStream('metadata-temp.tar.gz'));
```

works fine. But if I remove the `strictSSL: false` option (it defaults to `true`) then I get:
`Error: self signed certificate in certificate chain` and a 0 byte file is created.

To "fix" this for people in work environments like mine there would need to be some kind of config option that made calls using `[request](https://www.npmjs.com/package/request)` use the `strictSSL: false` option. The relevant code is [here](https://github.com/LightTable/LightTable/blob/53a90bee4c2c6758e4b5dffdffbc32256a05e00d/src/lt/objs/deploy.cljs#L75-76). I'll create a separate issue for this.

More relevant to this issue though is that @olange 's change to update the Node package `[request](https://www.npmjs.com/package/request)` to the latest version *does* fix proxy support.

**Comment by mluisbrown** [comment]
Just to confirm my issue beyond all doubt, if I hack the copy of `request` in my LT installation to force `strictSSL: false` then the Plugin manager in LT works fine over the office proxy :smile:

**Comment by djavanci** [comment]
I'm using LightTable 0.8.1 in Ubuntu 16.04 with CNTLM configured and in use by every other application on my machine. But I couldn't connect to server and get the plugins available. Reading this thread, I installed npm on my machine and checked the request version on path "lighttable-0.8.1-linux/resources/app/core:

$ npm list request
LightTable@0.8.1 ~/lighttable-0.8.1-linux/resources/app/core
└── request@2.14.0 

Then configured proxy on ntm with the commands:
npm config set proxy http://127.0.0.1:3128
npm config set https-proxy http://127.0.0.1:3128

and executed the command "npm update request" and after that, the command "npm list request" showed the following output:

LightTable@0.8.1 ~/lighttable-0.8.1-linux/resources/app/core
└── request@2.72.0 

After this, the Light Tabçle could connect to server and update the plugins.

**Comment by kenny-evitt** [comment]
@djavanci Thanks for confirming.

**Comment by olange** [comment]
Being back today at my customer's site, where I encountered the issue I reported on April 17, I built LightTable from the sources this time (rather than only updating the `request` module within the packaged version of LightTable), behind its enterprise proxy, and could confirm that updating the `request` module to the latest version indeed solved the issue of updating the plugins behind a proxy:

* build of LightTable 0.8.1 on Windows from sources (master branch as of today) fails refreshing the plugins list, nor update, nor install plugins;
* after update of the `request` module from version 2.14.0 to latest 2.72.0 within folder `deploy/core/`, and a new build from these updated sources, the new build of LightTable.exe allowed me to refresh the plugins list and install new plugins (I installed Markdown 0.0.3, which worked fine).

I would happily submit a PR with the changes, as it seems pretty sure from all comments, that it brings back proxy support, but I'm a bit confused about what the PR should contain. Could anyone guide me?

Obviously, the PR should contain the updated `deploy/core/package.json`, but `git status` shows quite a lot of deleted, updated and untracked files in `deploy/core/node_modules/`. Some of them are updated dependencies of the `request` module, but others were already displayed by `git st` as being updated or untracked, just after a `npm install` on a clean checkout of the sources, and before updating the request `module`.

**Comment by kenny-evitt** [comment]
@olange I submitted PR #2208 already but the core team (excluding me) is not sold on merging those changes. As-of now, at least one of the core team needs to reproduce the original issue. We haven't gotten around to doing so yet.

**Comment by Astara** [comment]
What is keeping proxy support from being added back in?  I'm behind a proxy and some things won't connect (like extensions).  

It's a simple proxy --- no authorization needed... just host:port in my browser.  Is that so complicated?

As for duplicating the env on your local machine... download squid from squid-cache.org.
It's not hard to build a basic version.   Would that duplicate the env you need?

Using a VirtualBox to create an isolated system would be another option if you need to ensure connectivity isolation...

tnx,
-linda

**Comment by olange** [comment]
@Astara Try the workaround I described in [my above comment of 17.04.2016](#issuecomment-211029093). Assuming you have `npm` on your command-line, this recipe might solve your issue; replace `‹lighttable-dir›` with the path to the folder you installed LightTable into:

```
> cd ‹lighttable-dir›\resources\app\core
> npm update request
[… snip snip …]
LightTable@0.8.1
`-- request@2.72.0
```

**Comment by olange** [comment]
@kenny-evitt proposed a PR #2208 to solve the issue and took the time to discuss it with core developers (many thanks by the way, I forgot to answer when I first read it last year), but the request was rejected because another request of the core developers is not fulfilled yet – they'd like to be able to test the proxy support in continuous integration tests. So our user requirement is subordinated to a tooling requirement…

I feel this is well blocked. And if I might add a comment: I understand well the QA concern, but I don't get why this improvement to the user experience, that eases adoption of LightTable in corporate environments, should be blocked by another improvement of the continuous build environment. 

Automating tests is very good practice – but if no one is able to setup the required tooling, then it is bad strategy to block the improvement. Users are very good at reporting issues, and we certainly would, if updating the `request` to its latest version had side-effects.

**Comment by Astara** [comment]
Olivier Lange wrote:
>
> @kenny-evitt <https://github.com/kenny-evitt> proposed a PR #2208 
> <https://github.com/LightTable/LightTable/pull/2208> to solve the 
> issue and took the time to discuss it with core developers (many 
> thanks by the way, I forgot to answer when I first read it last year), 
> but the request was rejected because another request of the core 
> developers is not fulfilled yet – they'd like to be able to test the 
> proxy support in continuous integration tests. So our user requirement 
> is subordinated to a tooling requirement…
>
----
    They can setup squid and test through that -- or to really isolate a 
client,
run it on a VM.

    I.e. the method is there.  If they want to set it up, they can.  But 
like you
say, if they aren't willing to automate the test, I'd think they weren't 
that
serious about automating -- and if using that as an excuse not to fix the
bug, then its pretty obvious they don't care to fix it.

Too bad.

    It's alot slower than FF, so I have a feeling the project is already 
on the decline.

**Comment by sbauer322** [comment]
Given that adding QA has been blocked by upgrading Light Table's internal ClojureScript to something remotely recent for at least a year, I have no qualms about reintroducing proxy support in the meantime.

>  I.e. the method is there. If they want to set it up, they can. But like you say, if they aren't willing to automate the test, I'd think they weren't that serious about automating -- and if using that as an excuse not to fix the bug, then its pretty obvious they don't care to fix it.

In a sense, you are correct @Astara. None of the active maintainers are pained by the proxy issue so no one has taken care of it. The project is currently being run entirely on people's free time... of which there is often little to none. Time spent on the project is almost always something that interests or bothers the maintainers. Short of having a commercial sponsor funding a maintainer for development, this pattern will continue.

As @olange alluded to, if someone can give more specific instructions (or even a link to a good quality guide) for setting up the squid proxy then it will lessen the amount of pain to get the proxy running to test the fix... which would improve the odds of it getting tested soon.

**Comment by Astara** [comment]
Scott Bauer wrote:
>
> Given that adding QA has been blocked by upgrading Light Table's 
> internal ClojureScript to something remotely recent for at least a 
> year, I have no qualms about reintroducing proxy support in the meantime.
>
>     I.e. the method is there. If they want to set it up, they can. But
>     like you say, if they aren't willing to automate the test, I'd
>     think they weren't that serious about automating -- and if using
>     that as an excuse not to fix the bug, then its pretty obvious they
>     don't care to fix it.
>
> In a sense, you are correct @Astara <https://github.com/Astara>. None 
> of the active maintainers are pained by the proxy issue so no one has 
> taken care of it. The project is currently being run entirely on 
> people's free time... of which there is often little to none. Time 
> spent on the project is almost always something that interests or 
> bothers the maintainers. Short of having a commercial sponsor funding 
> a maintainer for development, this pattern will continue.
>
----
    They would probably have to attract a large enough user-base for
a sponsor to see it as worthwhile. 

> As @olange <https://github.com/olange> alluded to, if someone can give 
> more specific instructions (or even a link to a good quality guide) 
> for setting up the squid proxy then it will lessen the amount of pain 
> to get the proxy running to test the fix... which would improve the 
> odds of it getting tested soon.
>
Do any of them run linux?  It's usually an included package in most
distros, which might be the easiest way.  I think even Cygwin (runs on
Windows) has squid in their list of offered packages. 

Instructions would vary by distro to some extent. 

Like Ubuntu, has a setup guide here... 
https://help.ubuntu.com/lts/serverguide/squid.html

I found that by typing in "how to setup squid proxy" to google
at the top entry.  If they have a specific platform, they'd add
that onto search line.

I'm not the best person to ask for simple instructions or a guide to
set it up, as I generate my squid proxy from source and have a
custom-setup for my home network.

The website @ http://www.squid-cache.org/ has pointers
to examples, a wiki, books and more, but the best way is
go for examples for whatever platform you are running
on and google on that.

Thinking about it -- I'm wondering -- the proxy issue only came
up when I scanned for extensions/plugins.  Does that go to a website
URL?  That might be the problem -- if it is going to some absolute port
and not using http(or https) as a transport, then they might have
more luck converting the extension-plugin scanner to http and web-requests,
but w/o delving into it more, not sure exactly what the problem is.

**Comment by mluisbrown** [comment]
> Thinking about it -- I'm wondering -- the proxy issue only came
up when I scanned for extensions/plugins.  Does that go to a website
URL?  That might be the problem -- if it is going to some absolute port
and not using http(or https) as a transport, then they might have
more luck converting the extension-plugin scanner to http and web-requests,
but w/o delving into it more, not sure exactly what the problem is.

The problem is only with getting plugin metadata, downloading plugins and for autoupdate. The relevante code is [here](https://github.com/LightTable/LightTable/blob/master/src/lt/objs/deploy.cljs#L74-86). It uses regular HTTPS web requests to the GitHub repo, see [here](https://github.com/LightTable/LightTable/blob/master/src/lt/objs/deploy.cljs#L32)

I already have an [open PR](https://github.com/LightTable/LightTable/pull/2214) to fix the proxy issues, including an option to disable strict SSL checking which is requires in some corporate environments. The PR also has detailed steps for being to able to test the fix.

In short, there are two issues:
* update the NPM `request` package to version 2.72.2 (or higher) as proxy support in general is broken in the version that LT is using (2.14.0).
* add an option to be able to disable strict SSL checking.

It is really unfortunate that a PR can be waiting to be merged for nearly a year just because QA can't test it in an automated way. Ignoring PRs is a great way to stop getting any community contributions 😞

**Comment by sbauer322** [comment]
@mluisbrown - thanks for pointing out your PR and the steps for being able to test the fix. Assuming Charles works for Windows and Linux (it should), I will give it a try this week.

I do agree it is quite unfortunate that there have been several important PRs stagnating, which is why we intend to address all outstanding PRs this year as part of [Light Table's 2017 Roadmap](http://lighttable.com/2017/01/27/light-table-roadmap-2017/).

**Comment by mluisbrown** [comment]
Thanks @sbauer322 for merging the PRs! I believe this issue can now be closed, finally 🎉 . I already closed #2212.

**Comment by sbauer322** [comment]
Closing as PRs #2214 and #2331 have been merged and should address this.
